assets/src/block-editor/plugins/wrapped-amp-preview-button.js

```diff
@@ -19,74 +13,43 @@ import AmpPreviewButton from '../components/amp-preview-button';
 /**
  * A wrapper for the AMP preview button that renders it immediately after the 'Post' preview button, when present.
  */
-class WrappedAmpPreviewButton extends Component {
-	/**
-	 * Constructs the class.
-	 *
-	 * @param {*} args Constructor arguments.
-	 */
-	constructor( ...args ) {
-		super( ...args );
-
-		this.root = document.createElement( 'div' );
-		this.root.className = 'amp-wrapper-post-preview';
-
-		this.postPreviewButton = document.querySelector( `.${ POST_PREVIEW_CLASS }` );
-	}
-
-	/**
-	 * Invoked immediately after a component is mounted (inserted into the tree).
-	 */
-	componentDidMount() {
-		if ( ! this.postPreviewButton ) {
-			return;
+function WrappedAmpPreviewButton() {
+	const root = useRef();
+	const postPreviewButton = useRef();
+
+	const isViewable = useSelect(
+		( select ) => select( 'core' ).getPostType( select( 'core/editor' ).getEditedPostAttribute( 'type' ) )?.viewable,
+		[],
+	);
+
+	useEffect( () => {
+		if ( isViewable && ! root.current && ! postPreviewButton.current ) {
+			postPreviewButton.current = document.querySelector( `.${ POST_PREVIEW_CLASS }` );
+
+			// Insert the AMP preview button immediately after the post preview button.
+			if ( postPreviewButton.current ) {
+				root.current = document.createElement( 'div' );
+				root.current.className = 'amp-wrapper-post-preview';
+				postPreviewButton.current.parentNode.insertBefore( root.current, postPreviewButton.current.nextSibling );
+			}
 		}
 
-		// Insert the AMP preview button immediately after the post preview button.
-		this.postPreviewButton.parentNode.insertBefore( this.root, this.postPreviewButton.nextSibling );
-	}
+		return () => {
+			if ( postPreviewButton.current && root.current ) {
+				postPreviewButton.current.parentNode.removeChild( root.current );
```

Comment:
Should this also set `postPreviewButton.current` and `root.current` to null?

Reply:
Or rather, just `root.current`?

Reply:
Or yeah, both right? As otherwise if it gets re-rendered then the logic above will never run since `! root.current && ! postPreviewButton.current`.

Reply:
Yes, they should be nulled here.

Reply:
This has been addressed in 9f3e1c8.